response_operations_ui/views/collection_exercise.py

```diff
@@ -1194,13 +1200,41 @@ def view_ci_versions(short_name: str, period: str, form_type: str) -> str:
 @login_required
 def save_ci_versions(short_name: str, period: str, form_type: str):
     ci_version = request.form.get("ci-versions")
+    ce_details = build_collection_exercise_details(short_name, period, include_ci=True)
     if "nothing-selected" == ci_version:
-        ce_details = build_collection_exercise_details(short_name, period, include_ci=True)
         collection_instrument_controllers.delete_registry_instruments(
             ce_details["collection_exercise"]["id"], form_type
         )
         return redirect(url_for("collection_exercise_bp.view_sample_ci_summary", short_name=short_name, period=period))
-    return redirect(url_for("collection_exercise_bp.view_collection_exercise", short_name=short_name, period=period))
+    else:
+        redis_cache = RedisCache()
+        survey_ref = redis_cache.get_survey_by_shortname(short_name).get("surveyRef")
+        list_of_cir_metadata_objects = None
+        if survey_ref:
+            try:
+                list_of_cir_metadata_objects = redis_cache.get_cir_metadata(survey_ref, form_type)
+            except ExternalApiError as e:
+                logger.info("Error Retrieving CIR metadata", survey_ref=survey_ref, form_type=form_type, error=e)
+
+            cir_metadata_object = next(
```

Comment:
I assume by this that there is more than one coming back from redis_cache.get_cir_metadata

Reply:
Yes, its the list of CIR version objects as per the response from the CIR service itself.